### PR TITLE
Task 86783 allow topic

### PR DESCRIPTION
**Pull request by vicpada**
Allow to pass topic name to topic sender and subscriber

### CODE REVIEW

File: src/Eshopworld.Messaging/Messenger.cs [hunk 1]
```diff
@@ -61,94 +62,117 @@ public Messenger([NotNull]string connectionString, [NotNull]string subscriptionI
         public async Task Send<T>(T message)
             where T : class
         {
-            if (!ServiceBusAdapters.ContainsKey(typeof(T)))
+            if (!ServiceBusAdapters.ContainsKey(GetTypeName<T>()))
```
Comment (ghost) [inline src/Eshopworld.Messaging/Messenger.cs]: We can refactor to save `GetTypeName` response in a variable here.

File: src/Eshopworld.Messaging/TopicAdapter.cs [hunk 2]
```diff
@@ -180,5 +183,29 @@ protected override void Dispose(bool disposing)
...
+                    $@"You can't create queues for the topic name {topicName} because it exceeds 260 characters.");
+            }
+        }
+
+        private static string CheckTopicType()
+        {
+            var topicType = typeof(T);
+            var topicName = topicType.GetEntityName();
+
+            if (topicName.Length > 260) // SB quota: Entity path max length
```
Comment (ghost) [inline src/Eshopworld.Messaging/TopicAdapter.cs:204]: We are repeating this topic name length check in these 2 methods. We can change `CheckTopicType` to `GetTopicName` and remove the length check from it. Then we can have length check only in `CheckTopicName` method

File: src/Tests/Eshopworld.Messaging.Tests/MessengerTopicTest.cs [hunk 3]
```diff
@@ -79,14 +92,64 @@ public async Task Test_ReceiveCreatesTheTopic()
...
+        const string topicName = "IdleTopicTest";
+        const int idleTimeMin = 5;
+
+        using (IDoFullMessaging msn = new Messenger(ServiceBusFixture.ConfigSettings.ServiceBusConnectionString, ServiceBusFixture.ConfigSettings.AzureSubscriptionId))
+        {
+            const string subscriptionName = "IdleTopicSubscription";
+            await msn.Subscribe<TestMessage>(_ => { }, subscriptionName, topicName, deleteOnIdleDurationInMinutes: idleTimeMin);
+            msn.CancelReceive<TestMessage>(topicName);
+            ServiceBusFixture.ServiceBusNamespace.AssertSingleTopicSubscriptionExists(topicName, subscriptionName);
+            await Task.Delay(TimeSpan.FromMinutes(6));
```
Comment (ghost) [inline src/Tests/Eshopworld.Messaging.Tests/MessengerTopicTest.cs:149]: We should not have such delays in our tests but I can see this pattern is used in many of our tests. We may need to introduce a separate test category like "LongRunning" to separate such tests from other unit tests.
For now, we may at least reduce it by some seconds to save time :) as it needs to be just over 5 mins here.

File: src/Eshopworld.Messaging/TopicAdapter.cs (hunk 2 above)
Comment (vicpada) [reply src/Eshopworld.Messaging/TopicAdapter.cs:204]: yes, but the error messages are different. That's why I kept both checks

File: src/Eshopworld.Messaging/Messenger.cs (hunk 1 above)
Comment (vicpada) [reply src/Eshopworld.Messaging/Messenger.cs]: agreed & updated

File: src/Tests/Eshopworld.Messaging.Tests/MessengerTopicTest.cs (hunk 3 above)
Comment (vicpada) [reply src/Tests/Eshopworld.Messaging.Tests/MessengerTopicTest.cs:149]: I tested locally with 5m10s and 5m20s and it didn't work. So I think we can stick to the 6 min. I agree it is not ideal. But 5 min is the min you can set for autodelete idle subscriptions.

**Review by ghost — APPROVED** [review]

**Pull request merged**